pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/dictionary/DoubleMutableDictionary.java

```diff
@@ -155,20 +181,70 @@ public boolean inRange(String lower, String upper, int indexOfValueToCompare, bo
 
   @Override
   public String getStringValue(int dictionaryId) {
-    return ((Double) getRawValueFromBiMap(dictionaryId)).toString();
-  }
-
-  private double getDouble(int dictionaryId) {
-    return (Double) dictionaryIdBiMap.get(dictionaryId);
+    return String.valueOf(_dictionaryIdBiMap.getValue(dictionaryId));
   }
 
   @Override
   public Object getMinVal() {
-    return min;
+    return _min;
   }
 
   @Override
   public Object getMaxVal() {
-    return max;
+    return _max;
+  }
+
+  @Override
+  public void print() {
+        System.out.println("************* printing dictionary for column : " + _column + " ***************");
+    for (int key : _dictionaryIdBiMap.keySet()) {
+      System.out.println(key + "," + _dictionaryIdBiMap.getValue(key));
+    }
+    System.out.println("************************************");
+  }
+
+  public static class Double2IntBiMap {
```

Comment:
Make it private or package-private. It's not "proper" map the way it's implemented